app/services/send_apply_again_call_to_action.rb

```diff
@@ -0,0 +1,22 @@
+# This worker will be as a one-off task to inform previously unsuccessful candidates about apply again
+class SendApplyAgainCallToAction
+  include Sidekiq::Worker
+
+  def perform
+    unsuccessful_application_forms.find_each do |application_form|
+      CandidateMailer.apply_again_call_to_action(application_form).deliver
+    end
+  end
+
+private
+
+  def unsuccessful_application_forms
+    ApplicationForm
+      .joins(:application_choices)
+      .where
+      .not(application_choices: {
+        status: ApplicationStateChange.valid_states - ApplicationStateChange::UNSUCCESSFUL_END_STATES.map(&:to_sym),
```

Comment:
Can we just query for `where(application_choices: { status: ApplicationStateChange::UNSUCCESSFUL_END_STATES })` ?

Reply:
The intention here is to get applications (forms) for which all the application choices have ended unsuccessfully. So if a candidate got rejected for one choice and accepted an offer for another (or just hasn't heard yet) then we shouldn't be sending this email.